src/pages/InputDemo/InputDemo.jsx

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-console */
+import React from 'react';
+import { TextField, SelectField, RadioField } from '../../components';
+import { selectOptions, radioOptionsCricket, radioOptionsFootball } from '../../config/constants';
+
+class InputDemo extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      sport: '',
+      cricket: '',
+      football: '',
+    };
+  }
+
+    handleNameChange = (e) => {
+      this.setState({ name: e.target.value }, () => {
+        console.log(this.state);
+      });
+    }
+
+    handleSportChange = (e) => {
+      this.setState({ sport: e.target.value }, () => console.log(this.state));
+      if (e.target.value === 'Select') {
+        this.setState({ sport: '' });
+      }
+      return e.target.value === 'cricket' ? this.setState({ football: '' }) : this.setState({ cricket: '' });
+    }
+
+    handlePositionChange = (e) => {
+      const { sport } = this.state;
+      return sport === 'cricket' ? this.setState({ cricket: e.target.value }, () => console.log(this.state)) : this.setState({ football: e.target.value }, () => console.log(this.state));
+    }
+
+      RadioOption = () => {
+        let { radioValue } = this.state;
+        const { sport } = this.state;
+        if (sport === 'cricket') {
+          radioValue = radioOptionsCricket;
+        } else if (sport === 'football') {
+          radioValue = radioOptionsFootball;
+        }
+        return (radioValue);
+      };
+
```

Comment:
never name function like this. optimize code and remove if-else.

Reply:
remove functions, if-else and optimize code as per the requirement.